tests/e2e-playwright/pages/admin/post-type/index.ts

```diff
@@ -55,33 +55,54 @@ export default class PostType {
 		return new QueryLoop( this.queryLoopPatternSelection, this.page );
 	}
 
-	async getPreviewURL(): Promise< string > {
-		const params = new URL( await this.page.url() ).searchParams;
+	async goToPreview(): Promise< Page > {
+		await this.page.locator( 'button:has-text("Preview")' ).first().click();
 
-		return `/?page_id=${ params.get( 'post' ) }`;
+		const [ previewPage ] = await Promise.all( [
+			this.page.waitForEvent( 'popup' ),
+			this.page.locator( 'text=Preview in new tab' ).click(),
+		] );
+		await previewPage.waitForLoadState();
+		return previewPage;
 	}
 
-	async publish(): Promise< Response > {
+	async viewPage(): Promise< Page > {
 		await this.page
-			.locator( '[aria-label="Editor top bar"] >> text=Publish' )
+			.locator( '[aria-label="Editor publish"]' )
+			.locator( 'text=View Page' )
 			.click();
+		return this.page;
```

Comment:
Not completely clear for me: we changed `this.page` here, so now methods and properties of the object won't work?
Also `this.page` is the reference to the `page` we got in our test?
If so, I just think that our objects become "stale" when we navigate to another page, but we don't have any sign about that. It looks like our page abstractions are leaky and deceptive. 
As a solution, I though about a flag in page objects that they are still actual (mark them stale when navigate to another page, or we need to check current URL for the same purpose. And we need to return not `this.page` but a new page object in cases like this one. WDYT?

Reply:
@merkushin perfect comments!

I will merge to fix the pipeline and we can discuss it more and we can open a new PR with some improvements on our page model.